app/page.js

```diff
@@ -1,41 +1,74 @@
 "use client";
 
-import React, { useState, useEffect, useRef } from "react";
+import React, { useState, useContext } from "react";
 
-import { spawnNode, Network, NodeConfig } from 'lumina-node';
+import { Network, NodeConfig } from "lumina-node";
+
+import { LuminaContext, LuminaContextProvider } from "./lumina";
 
 export default function Home() {
-  let initialized = useRef(false);
-  let [node, setNode] = useState(null);
+  return (
+    <LuminaContextProvider>
+      <Page />
+    </LuminaContextProvider>
+  );
+}
 
-  useEffect(() => {
-    const init = async () => {
-      let node = await spawnNode();
-      setNode(node);
+function Page() {
+  const node = useContext(LuminaContext);
+
+  const [peerTrackerInfo, setPeerTrackerInfo] = useState();
+  const [syncerInfo, setSyncerInfo] = useState();
+
+  const statsUpdateInterval = setInterval(() => {
+    const update = async () => {
+      if (node && (await node.isRunning())) {
+        setPeerTrackerInfo(await node.peerTrackerInfo());
+        setSyncerInfo(await node.syncerInfo());
+      }
     };
-    if (!initialized.current) {
-      initialized.current = true;
-      init();
-    }
-  }, [node]);
+    update();
+  }, 1000);
```

Comment:
this will create new intervals each time the component is re-rendered, so each time peer tracker and syncer info are updated. Instead you should wrap it in useEffect with empty deps list to run just once, and a cleanup callback
```js
useEffect(() => {
  const interval = setInterval(...);
  return () => clearInterval(interval);
}, [])
```

Reply:
actually dependency list should include `[node]`

Reply:
You're right, haven't thought about the side effects. Fixed :+1: